File: packages/react-native/ReactAndroid/src/test/java/com/facebook/testutils/shadows/ShadowReadableNativeArray.java
/*
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

package com.facebook.testutils.shadows;

import androidx.annotation.NonNull;
import com.facebook.react.bridge.Dynamic;
import com.facebook.react.bridge.JavaOnlyArray;
import com.facebook.react.bridge.ReadableArray;
import com.facebook.react.bridge.ReadableMap;
import com.facebook.react.bridge.ReadableNativeArray;
import com.facebook.react.bridge.ReadableType;
import java.util.ArrayList;
import org.robolectric.annotation.Implementation;
import org.robolectric.annotation.Implements;

@Implements(ReadableNativeArray.class)
public class ShadowReadableNativeArray extends ShadowNativeArray implements ReadableArray {

  protected JavaOnlyArray backingArray;

  @Implementation
  protected void __constructor__() {
    this.backingArray = new JavaOnlyArray();
  }

  @Implementation
  public int size() {
    return this.backingArray.size();
  }

  @Implementation
  public boolean isNull(int index) {
    return this.backingArray.isNull(index);
  }

  @Implementation
  public boolean getBoolean(int index) {
    return this.backingArray.getBoolean(index);
  }

  @Implementation
  public double getDouble(int index) {
    return this.backingArray.getDouble(index);
  }

  @Implementation
  public int getInt(int index) {
    return this.backingArray.getInt(index);
  }

  @Implementation
  public long getLong(int index) {
    return this.backingArray.getLong(index);
  }

  @Implementation
  public @NonNull String getString(int index) {
    return this.backingArray.getString(index);
  }

  @Implementation
  public @NonNull ReadableArray getArray(int index) {
    return this.backingArray.getArray(index);
  }

  @Implementation
  public @NonNull ReadableMap getMap(int index) {
    return this.backingArray.getMap(index);
  }

  @Implementation
  public @NonNull Dynamic getDynamic(int index) {
    return this.backingArray.getDynamic(index);
  }

  @Implementation
  public @NonNull ReadableType getType(int index) {
    return this.backingArray.getType(index);
  }

  @Implementation
  public @NonNull ArrayList<Object> toArrayList() {
    return this.backingArray.toArrayList();
  }
}

File: packages/react-native/ReactAndroid/src/test/java/com/facebook/testutils/shadows/ShadowReadableNativeArray.kt
/*
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

package com.facebook.testutils.shadows

import com.facebook.react.bridge.Dynamic
import com.facebook.react.bridge.ReadableArray
import com.facebook.react.bridge.ReadableMap
import com.facebook.react.bridge.ReadableNativeArray
import com.facebook.react.bridge.ReadableType
import java.util.ArrayList
import org.robolectric.annotation.Implementation
import org.robolectric.annotation.Implements

@Implements(ReadableNativeArray::class)
open class ShadowReadableNativeArray : ShadowNativeArray(), ReadableArray {

  @Implementation override fun size(): Int = backingArray.size()

  @Implementation override fun isNull(index: Int): Boolean = backingArray.isNull(index)

  @Implementation override fun getBoolean(index: Int): Boolean = backingArray.getBoolean(index)

  @Implementation override fun getDouble(index: Int): Double = backingArray.getDouble(index)

  @Implementation override fun getInt(index: Int): Int = backingArray.getInt(index)

  @Implementation override fun getLong(index: Int): Long = backingArray.getLong(index)

  @Implementation override fun getString(index: Int): String? = backingArray.getString(index)

  @Implementation override fun getArray(index: Int): ReadableArray? = backingArray.getArray(index)

  @Implementation override fun getMap(index: Int): ReadableMap? = backingArray.getMap(index)

  @Implementation override fun getDynamic(index: Int): Dynamic = backingArray.getDynamic(index)

  @Implementation override fun getType(index: Int): ReadableType = backingArray.getType(index)

  @Implementation override fun toArrayList(): ArrayList<Any?> = backingArray.toArrayList()
}
